### PR TITLE
v3.3.0-beta.20: Typing updates

### PR DESCRIPTION
**Pull request by parties**
The main changes are to how the library exports itself.

I moved away from the TypeScript namespace model and back to a class-based model (namespaces are no longer recommended in modern TypeScript).

I've also added some strong types for event handlers so clients know exactly what is expected of them _and_ what data is coming back when the event fires.

### CODE REVIEW

File: src/post-robot.d.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  interface FireAndForgetRequestOptionsType {
+    domain?: DomainMatcher | undefined;
+    fireAndForget?: true | undefined;
+    timeout?: number | undefined;
+  }
+
+  export type HandlerType = (params: {
+    source: CrossDomainWindowType;
+    origin: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (PauloAtSparkCognition) [inline src/post-robot.d.ts:28]: Is there a way for future use we can add what type of data it is?

**Review by PauloAtSparkCognition — APPROVED** [review]
LGTM :rocket

File: src/post-robot.d.ts (hunk 1 above)
Comment (parties) [reply src/post-robot.d.ts:28]: Yeah we could most likely add a generic type to it, allowing the consumer to specify the data type and even the return type.

Comment (parties) [reply src/post-robot.d.ts]: Just pushed a change to allow for specifying the type of the event handler here

**Pull request merged**